src/server/_security.py

```diff
@@ -0,0 +1,314 @@
+from typing import Dict, List, Optional, Set, cast
+from enum import Enum
+from datetime import date, timedelta
+from functools import wraps
+from uuid import uuid4
+from flask import g, Response
+from flask_limiter import Limiter
+from flask_limiter.util import get_remote_address
+from werkzeug.local import LocalProxy
+from sqlalchemy import Table, Column, String, Integer, Boolean
+from ._config import API_KEY_REQUIRED_STARTING_AT, RATELIMIT_STORAGE_URL, URL_PREFIX
+from ._common import app, request, db
+from ._exceptions import MissingAPIKeyException, UnAuthenticatedException
+from ._db import metadata, TABLE_OPTIONS
+from ._logger import get_structured_logger
+import re
+
+API_KEY_HARD_WARNING = API_KEY_REQUIRED_STARTING_AT - timedelta(days=14)
+API_KEY_SOFT_WARNING = API_KEY_HARD_WARNING - timedelta(days=14)
+
+API_KEY_WARNING_TEXT = "an api key will be required starting at {}, go to https://delphi.cmu.edu to request one".format(API_KEY_REQUIRED_STARTING_AT)
+
+user_table = Table(
+    "api_user",
+    metadata,
+    Column("id", Integer, primary_key=True, autoincrement=True),
+    Column("api_key", String(50), unique=True),
+    Column("email", String(100)),
+    Column("roles", String(255)),
+    Column('tracking', Boolean(), default=True),
+    Column('registered', Boolean(), default=False),
+    **TABLE_OPTIONS,
+)
+
+
+class DBUser:
+    id: int
+    api_key: str
+    email: str
+    roles: Set[str]
+    tracking: bool = True
+    registered: bool = False
```

Comment:
This class is just a wrapper to interact with `api_user` table thats why we have that `user_table` declared above and DBUser class with all the fields definition. It can be probably rewritten to use `db` object to interact with that table inside that class, but I'm not sure if it's worth it. The structure will remain the same and we will need to rewrite all methods to use `db` object.

Reply:
Added parent class for DBUser and APIUser, but left fields declaration as it is.